packages/edge/src/middleware-helpers.ts

```diff
@@ -1,9 +1,39 @@
+export interface ModifiedRequest {
+  /**
+   * If this is set, the request headers will be overridden with this value.
+   */
+  headers?: Headers;
+}
+
 export interface ExtraResponseInit extends Omit<ResponseInit, 'headers'> {
   /**
    * These headers will be sent to the user response
    * along with the response headers from the origin.
    */
   headers?: HeadersInit;
+  /**
+   * These fields will override the request from clients.
```

Comment:
```suggestion
   * Fields to rewrite for the upstream request
```

<picture data-single-emoji=":maybe-test:" title=":maybe-test:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..VPSCGzkV6wwP5RXS.TysfFeIjQCvMXJOzio6ahuL7XK2G4lFJ0KP4_cC0BN0M3IsflkyFnDIlof3KpfZWukCqYsEVYWDJl0NK0qYATmytx1VAp5OwTtwuiT9jb_PmI1673z_6uG-vsdU.jwu2zVTQcy9xclngZ7O8gA" alt=":maybe-test:" align="absmiddle"></picture> ? I am not good with lingo maybe @molebox can give us a hand? <picture data-single-emoji=":eyeses:" title=":eyeses:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..1u_9G79noPw1oB9N.BxHmvFM2rRvGM1QbPPmww6E3ZNf5npeGz3-fXf-7qdb27vBrinaSkGxQBKyG-0MfKxC2KP76vLlgxyx784G9KyKjhTUJG75zWWtn-KE2vq7vZkxrWULtyA.BOq0CvvncgVwsaaN0xhc3g" alt=":eyeses:" align="absmiddle"></picture>

Reply:
lgtm